src/contracts/core/AllocationManager.sol

```diff
@@ -750,16 +750,22 @@ contract AllocationManager is
             uint256[] memory indices =
```

Comment:
uh can u add a todo here to not load the array into memory? not sure the exact behavior under the hood here actually